pupil_src/shared_modules/calibration_choreography/base_plugin.py

```diff
@@ -0,0 +1,400 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2020 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+import abc
+import enum
+import logging
+import typing as T
+
+from pyglui import ui
+from plugin import Plugin
+
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Move to gazer module
+class Gazer(Plugin, abc.ABC):
+    label = None
+
+    @staticmethod
+    def registered_gazers() -> T.Mapping[str, "Gazer"]:
+        return dict(Gazer.__registered_choreography_plugins)
+
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__(*args, **kwargs)
+        try:
+            store = Gazer.__registered_choreography_plugins
+        except AttributeError:
+            Gazer.__registered_choreography_plugins = {}
+            store = Gazer.__registered_choreography_plugins
+        assert isinstance(cls.label, str), f"Gazer subclass {cls.__name__} must overwrite string class property \"label\""
+        store[cls.label] = cls
+
+
+# TODO: Move to gazer module
+class Gazer2D(Gazer):
+    label = "2D"
+
+
+# TODO: Move to gazer module
+class Gazer3D(Gazer):
+    label = "3D"
+
+
+class GazeDimensionality(enum.Enum):
+    GAZE_2D = "2d"
+    GAZE_3D = "3d"
+
+
+class ChoreographyMode(enum.Enum):
+    CALIBRATION = "calibration"
+    ACCURACY_TEST = "accuracy_test"
+
+    @property
+    def label(self) -> str:
+        return self.value.replace("_", " ").title()
+
+
+class ChoreograthyAction(enum.Enum):
```

Comment:
@romanroibu 
I was just having a look at the new stuff and noticed a typo here :P
Also at the class below.
Probably should do a ctrl-f for 'Choreograthy' on your feature-dev branch

Reply:
great catch, thanks!